docs/java-api/client.asciidoc

```diff
@@ -173,11 +173,21 @@ Client client = TransportClient.builder().settings(settings).build();
 
 Or using `elasticsearch.yml` file as shown in <<node-client>>
 
-The client allows sniffing the rest of the cluster, which adds data nodes
-into its list of machines to use. In this case, note that the IP addresses
-used will be the ones that the other nodes were started with (the
-"publish" address). In order to enable it, set the
-`client.transport.sniff` to `true`:
+The Transport client comes with a cluster sniffing feature which
+allows it to dynamically add new hosts and remove old ones.
+When sniffing is enabled the nodes specified during client construction via
+addTransportAddress are connected to first. After this, the client will search
+the cluster state for available data nodes, replacing its internal node list
```

Comment:
shall we replace "search the cluster state" with "call the cluster state api internally" or something along those lines, just to clarify how a client manages to inspect the cluster state?